packages/next/server/render.tsx

```diff
@@ -1266,7 +1291,7 @@ export async function renderToHTML(
 
       return {
         bodyResult,
-        documentElement: () => (Document as any)(),
+        documentElement: () => <Document {...({} as any)} />,
```

Comment:
I think we need to keep the function invocation for `Document` component? cc @devknoll

Reply:
Yeah currently it's intended that the functional document components aren't rendered with React, which is why they are just called as a function here.

Reply:
Thanks, I’ll check this again!